### PR TITLE
[nemo-qml-plugin-calendar] Detect orphan events. Contributes to MER#1247

### PR DESCRIPTION
**Pull request by chriadam**
This commit ensures that we don't expose "orphan" (aka "ghost") events
to the UI.  Instead, such events are now deleted after an appropriate
error is emitted for debugging purposes.

Contributes to MER#1247

### CODE REVIEW

**Comment by chriadam** [comment]
@pvuorela @blammit

**Comment by blammit** [comment]
LSTM

File: src/calendarworker.cpp [hunk 1]
```diff
@@ -553,12 +553,30 @@ void NemoCalendarWorker::loadData(const QList<NemoCalendarData::Range> &ranges,
...
         // of the event still exists.
-        if (mStorage->notebook(mCalendar->notebook(e)).isNull())
+        if (mStorage->notebook(mCalendar->notebook(e)).isNull()) {
+            // This may be a symptom of a deeper bug: if a sync adapter (or mkcal)
+            // doesn't delete events which belong to a deleted notebook, then the
+            // events will be "orphan" and need to be deleted.
+            if (mStorage->load(e->uid())) {
+                KCalCore::Incidence::Ptr orphan = mCalendar->incidence(e->uid(), KDateTime());
+                if (!orphan.isNull()) {
+                    if (mCalendar->deleteIncidence(orphan)) {
```
Comment (pvuorela) [inline src/calendarworker.cpp]: Looking at deleteIncidence(), it appears DeleteVisitor calls only deleteEvent(), not deleteEventInstances(), leaving recurrence-id instances behind. Compare to NemoCalendarWorker::deleteAll().

Comment (chriadam) [inline src/calendarworker.cpp]: Urgh.  Thank you.

**Comment by pvuorela** [comment]
LGTM

**Pull request merged**